### PR TITLE
Extended dialect triat to support numeric prefixed identifiers

### PR DESCRIPTION
**Pull request by groobyming**
Relates to  #1186. This PR includes changes to resolve the issue and enhance project functionality. Review appreciated.

### CODE REVIEW

File: src/tokenizer.rs [hunk 1]
```diff
@@ -800,7 +800,9 @@ impl<'a> Tokenizer<'a> {
 
                     // mysql dialect supports identifiers that start with a numeric prefix,
                     // as long as they aren't an exponent number.
-                    if dialect_of!(self is MySqlDialect | HiveDialect) && exponent_part.is_empty() {
+                    if (dialect_of!(self is MySqlDialect | HiveDialect) && exponent_part.is_empty())
+                        || self.dialect.supports_numeric_prefix()
```
Comment (alamb) [inline src/tokenizer.rs:804]: I wonder why the check for `self.dialect.supports_numeric_prefix()` doesn't also check `exponent_part.is_empty`? 

That would mean that tokens like `1e4` are going to be parsed as identifiers

What if you wrote this check like

```rust
  if self.dialect.supports_numeric_prefix() && exponent_part.is_empty()) {
```

And then changed `MySqlDialect` and `HiveDialect` to return true for `supports_numeric_prefix`?

File: src/dialect/mod.rs [hunk 2]
```diff
@@ -143,6 +143,10 @@ pub trait Dialect: Debug + Any {
     fn supports_start_transaction_modifier(&self) -> bool {
         false
     }
+    /// Returns true if the dialect supports identifiers that start with a numeric prefix
```
Comment (alamb) [inline src/dialect/mod.rs:146]: Could you please add an example of such an identifier? (e.g. `123Column`)?

**Review by alamb — COMMENTED** [review]
Thank you for this contribution @groobyming  -- I have a suggestion that I think would make the code cleaner and perhaps also work more like how you want

I apologize for the delay in reviewing

**Comment by alamb** [comment]
Marking as draft as I think this PR is no longer waiting on feedback. Please mark it as ready for review when it is ready for another look

File: src/tokenizer.rs (hunk 1 above)
Comment (groobyming) [reply src/tokenizer.rs:804]: Great suggestion! I overlooked this scenario, and I'll fix this logic issue

File: src/dialect/mod.rs (hunk 2 above)
Comment (groobyming) [reply src/dialect/mod.rs:146]: Our scenario will use numbers as business ids. The rule for table names is the business id plus an underscore followed by the specific table name. 
eg：59901_user_login

**Review by alamb — APPROVED** [review]
Thanks again @groobyming  -- I took the liberty of moving the check to be entirely trait based and merging up to resolve conflicts.

**Pull request merged**